### PR TITLE
79 web chat api backend

### PR DESCRIPTION
**Pull request by NastyaKolbina**

### CODE REVIEW

File: website/views.py [hunk 1]
```diff
@@ -40,21 +44,30 @@ def get_messages(request, *args, **kwargs):
...
+                message = {}
+                for item in data:
+                    if item in json_template:
+                        message[item] = data[item]
+                messages_list.append(message)
+            json_string = json.dumps(messages_list, ensure_ascii=False, indent=4, sort_keys=True, default=str)
+            return JsonResponse({"messages": json_string})
+        else:
+            return messages_list
+        # ToDo Anastasia: Retrieve here messages for the user with helpers.MessageHistoryHelpers.get_last_user_messages()
```
Comment (Ovaday) [inline website/views.py]: remove comments, they were only for you

File: website/views.py [hunk 2]
```diff
@@ -40,21 +44,30 @@ def get_messages(request, *args, **kwargs):
         return Http401()
     if request.method == "GET":
 
-        # ToDo Anastasia: Retrieve here messages for the user with helpers.MessageHistoryHelpers.get_last_user_messages()
-        # ToDo Anastasia: Include case when there are no messages.
-        # You should return list with the following format:
-        messages = [{
+        record_list = get_last_user_messages(user_id)
+        json_template = {
```
Comment (Ovaday) [inline website/views.py]: would be better to name as an 'output_template'

File: website/views.py [hunk 3]
```diff
@@ -69,7 +82,17 @@ def get_messages(request, *args, **kwargs):
 # No need to test if that works right now. We will check it later.
```
Comment (Ovaday) [inline website/views.py]: remove comments here that are for you

File: website/views.py [hunk 4]
```diff
@@ -69,7 +82,17 @@ def get_messages(request, *args, **kwargs):
...
 
 def insert_message(request, *args, **kwargs):
-    print(request)
+    user_id = kwargs.get('user_id')
+    message = kwargs.get('message')
+    username = kwargs.get('username')
+    if user_id > 0:
```
Comment (Ovaday) [inline website/views.py]: by default user_id is a string

**Review by Ovaday — CHANGES_REQUESTED** [review]
Fix comments, otherwise good work

File: website/views.py [hunk 5]
```diff
@@ -66,10 +78,20 @@ def get_messages(request, *args, **kwargs):
 #  length (minimum - 5, max - 1000). In addition, there should be non-null user_id.
 #  To insert, use methods from the same helper, construct_message and insert_web_message.
 #  As a result return JSON with response 200 - ok (google how it should look like.)
-# No need to test if that works right now. We will check it later.
+
 
 def insert_message(request, *args, **kwargs):
-    print(request)
+    user_id = int(kwargs.get('user_id'))
```
Comment (Ovaday) [inline website/views.py]: check for None first:
[http://127.0.0.1:8000/api/messages/](http://127.0.0.1:8000/api/messages/)

Should return error404 and not an exception.

**Review by Ovaday — CHANGES_REQUESTED** [review]
1 fix and ok

**Comment by NastyaKolbina** [comment]
done

вс, 30 апр. 2023 г. в 19:57, Evgenii ***@***.***>:

**Review by Ovaday — APPROVED** [review]

**Pull request merged**